src/software/amazon/ionschema/internal/TypeImpl.kt

```diff
@@ -27,34 +27,30 @@ internal class TypeImpl(
 
     init {
         var foundTypeConstraint = false
-        val tmpConstraints = ionStruct
+        constraints = ionStruct.asSequence()
                 .filter { it.fieldName == null || (schema.getSchemaSystem() as IonSchemaSystemImpl).isConstraint(it.fieldName) }
                 .map {
                     if (it.fieldName.equals("type")) {
```

Comment:
very nit: consider using `onEach` for code that simply inspects sequences

Reply:
good to know, but `map` seems appropriate here, as we're translating from fields in `ionStruct` to Constraint instances;  leaving as-is

Reply:
I should've been more specific. I was talking about moving only the `if` condition to `onEach` 

full example: 
```kotlin 
.onEach { if (it.fieldName.equals("type")) foundTypeConstraint = true }
.map { (schema.getSchemaSystem() as IonSchemaSystemImpl).constraintFor(it, schema) }                
```

but as I said, very nit don't mind leaving as is.

Reply:
aha, got it, thank you, i appreciate the additional clarify this provides; updated